addons/misra.py

```diff
@@ -316,7 +337,7 @@ def hasSideEffectsRecursive(expr):
             e = e.astOperand1
         if e and e.str == '.':
             return False
-    if expr.str in ('++', '--', '='):
+    if expr.str in {'++', '--'} | ASSIGNMENT_OPERATORS:
```

Comment:
This is fine.. but I wonder if you can add `expr.isAssignmentOp()` .. we have that method in the `Token` C++ class in lib.